validator/src/journal/publisher_ffi.rs

```diff
@@ -380,7 +381,7 @@ pub extern "C" fn block_publisher_on_chain_updated(
     let mut publisher = unsafe { (*(publisher as *mut BlockPublisher)).clone() };
     py.allow_threads(move || {
         publisher.publisher.on_chain_updated_internal(
-            chain_head,
+            chain_head.block(),
```

Comment:
That method is only used in the BlockPublisher unit tests. I haven't removed the use of blockwrappers from those tests, because this is just a chain controller PR.

Reply:
It will get fixed in the publisher PR, so no worries